marshal.go

```diff
@@ -74,15 +75,28 @@ func (self ReplayUnmarshaller) Unmarshal(
 	for _ = range vql.Eval(context.Background(), scope) {
 	}
 
-	// Return nil here indicates not to set the value into the
-	// scope (since we already did in the Replay above).
-	return nil, nil
+	return scope, nil
```

Comment:
This should return nil so it is not set in the scope. LET queries already set the scope.